src/node_embedding_api.h

```diff
@@ -0,0 +1,484 @@
+    NAPI_CDECL* node_embedding_runtime_configure_callback)(
+    void* cb_data,
+    node_embedding_platform platform,
+    node_embedding_runtime_config runtime_config);
+
+// The error is to be handled by using napi_env.
+typedef void(NAPI_CDECL* node_embedding_runtime_preload_callback)(
+    void* cb_data,
+    node_embedding_runtime runtime,
+    napi_env env,
+    napi_value process,
+    napi_value require);
+
+// The error is to be handled by using napi_env.
+typedef napi_value(NAPI_CDECL* node_embedding_runtime_loading_callback)(
+    void* cb_data,
+    node_embedding_runtime runtime,
+    napi_env env,
+    napi_value process,
+    napi_value require,
+    napi_value run_cjs);
+
+// The error is to be handled by using napi_env.
+typedef void(NAPI_CDECL* node_embedding_runtime_loaded_callback)(
+    void* cb_data,
+    node_embedding_runtime runtime,
+    napi_env env,
+    napi_value load_result);
+
+// The error is to be handled by using napi_env.
+typedef napi_value(NAPI_CDECL* node_embedding_module_initialize_callback)(
+    void* cb_data,
+    node_embedding_runtime runtime,
+    napi_env env,
+    const char* module_name,
+    napi_value exports);
+
+typedef node_embedding_status(NAPI_CDECL* node_embedding_task_run_callback)(
+    void* cb_data);
+
+typedef node_embedding_status(NAPI_CDECL* node_embedding_task_post_callback)(
+    void* cb_data,
+    node_embedding_task_run_callback run_task,
+    void* task_data,
+    node_embedding_data_release_callback release_task_data,
+    bool* is_posted);
+
+// The error is to be handled by using napi_env.
+typedef void(NAPI_CDECL* node_embedding_node_api_run_callback)(void* cb_data,
+                                                               napi_env env);
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+EXTERN_C_START
+
+//------------------------------------------------------------------------------
+// Error handling functions.
+//------------------------------------------------------------------------------
+
+// Gets the last error message for the current thread.
+NAPI_EXTERN const char* NAPI_CDECL node_embedding_last_error_message_get();
+
+// Sets the last error message for the current thread.
+NAPI_EXTERN void NAPI_CDECL
+node_embedding_last_error_message_set(const char* message);
+
+// Sets the last error message for the current thread using C printf string
+// formatting.
+NAPI_EXTERN void NAPI_CDECL
+node_embedding_last_error_message_set_format(const char* format, ...);
+
+//------------------------------------------------------------------------------
+// Node.js global platform functions.
+//------------------------------------------------------------------------------
+
+// Runs Node.js main function.
+// By default it is the same as running Node.js from CLI.
+NAPI_EXTERN node_embedding_status NAPI_CDECL node_embedding_main_run(
```

Comment:
Same comment as in docs, not sure this is a platform function versus a helper function to avoid having to call methods from the different categoriies ?